### ISSUE TITLE
Transaction Management

### ISSUE THREAD

**Issue by cressie176** [romeerez/orchid-orm#22]
Hi @romeerez,

Thanks for orchid-orm - I'm keeping a close eye on it as I agree with your assessment of other popular ORMs. One of my concerns with prisma is that transaction management seems very much an after thought, which is worrying given it's a database library. I like how you've incorporated transactions into the fluent API but, depending on how the application's code is organised this will require passing the transaction variable to function calls.

Sequelise [solves](https://sequelize.org/docs/v6/other-topics/transactions/#automatically-pass-transactions-to-all-queries) this quite neatly using [cls](https://github.com/Jeff-Lewis/cls-hooked) or [async local storage](https://nodejs.org/dist/latest-v18.x/docs/api/async_hooks.html#class-asynclocalstorage). I wondered if you have plans to do the same?

I have a second question, which is how you specify different isolation levels. It doesn't appear possible with the current API.

**Comment by romeerez** [comment]
Currently, no support for async local storage or even for isolation levels.

Thank you for pointing, I'm putting this on the to-do list.

**Comment by romeerez** [comment]
Ready! [Here are the docs](https://orchid-orm.netlify.app/guide/query-builder-transactions.html)

Sequelize supports manual, automatic, + implicitly passing transactions via [cls](https://github.com/Jeff-Lewis/cls-hooked) package.

But Orchid ORM can afford breaking changes, so I removed the previous way of transaction and made `AsyncLocalStorage` to be the default and the only option. It happens implicitly under the hood, so users may not worry about it.

Added support for customizing isolation level, it can be passed to a `$transaction` as a first string argument.
